dependencydiff/dependencydiff.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dependencydiff
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/ossf/scorecard/v4/checks"
+	"github.com/ossf/scorecard/v4/clients"
+	"github.com/ossf/scorecard/v4/clients/githubrepo"
+	"github.com/ossf/scorecard/v4/clients/githubrepo/roundtripper"
+	"github.com/ossf/scorecard/v4/pkg"
+)
+
+// GetDependencyDiffResults gets dependency changes between two given code commits BASE and HEAD
+// along with the Scorecard check results of the dependencies, and returns a slice of DependencyCheckResult.
+func GetDependencyDiffResults(ownerName, repoName, baseSHA, headSHA string) ([]pkg.DependencyCheckResult, error) {
+	ctx := context.Background()
+	// Fetch dependency diffs using the GitHub Dependency Review API.
+	deps, err := FetchDependencyDiffData(ctx, ownerName, repoName, baseSHA, headSHA)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching dependency changes: %w", err)
+	}
+
+	ghRepo, err := githubrepo.MakeGithubRepo(path.Join(ownerName, repoName))
+	if err != nil {
+		return nil, fmt.Errorf("error creating the github repo: %w", err)
+	}
+	ghRepoClient := githubrepo.CreateGithubRepoClient(ctx, nil)
+	ossFuzzRepoClient, err := githubrepo.CreateOssFuzzRepoClient(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating the oss fuzz repo client: %w", err)
+	}
+	vulnsClient := clients.DefaultVulnerabilitiesClient()
+
+	results := []pkg.DependencyCheckResult{}
+	for _, d := range deps {
+		if d.SourceRepository != nil {
+			if *d.SourceRepository == "" {
```

Comment:
How about appending a result with empty scorecard results, to make it explicit we could not find the result?
This way the caller can decide what do do with it: ignore it, or display a message to the user that it could not be retrieved (e.g., in the GHA).

Reply:
yes, i think it makes sense that we add empty scorecard results to those without a srcRepoURL.